src/service_discovery.rs

```diff
@@ -116,13 +116,32 @@ impl<Exe: Executor> ServiceDiscovery<Exe> {
             } = convert_lookup_response(&response)?;
             is_authoritative = authoritative;
 
-            // use the TLS connection if available
-            let connection_url = if let Some(u) = &broker_url_tls {
```

Comment:
Why `broker_url_tls` is not None when the server doesn't support tls? Is it possible to be a misconfiguration?

Reply:
The `broker_url_tls` on here is parsed from server which the old client logic used to connect to server even if the configuration is non-tls url. It makes some trouble such as certificate verification problem even user not want to use tls.

Reply:
I was thinking if it's possible there is a server configuration causes the server to send the tls url even it doesn't handle tls. But that is not related to your change. All good, thank you!